server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java

```diff
@@ -692,35 +702,42 @@ private boolean isAppExpired(String appId) {
    * @param shuffleIds
    */
   public void removeResourcesByShuffleIds(String appId, List<Integer> shuffleIds) {
-    if (CollectionUtils.isEmpty(shuffleIds)) {
-      return;
-    }
+    Lock writeLock = getAppWriteLock(appId);
+    writeLock.lock();
+    try {
+      if (CollectionUtils.isEmpty(shuffleIds)) {
+        return;
+      }
 
-    LOG.info("Start remove resource for appId[{}], shuffleIds[{}]", appId, shuffleIds);
-    final long start = System.currentTimeMillis();
-    final ShuffleTaskInfo taskInfo = shuffleTaskInfos.get(appId);
-    if (taskInfo != null) {
-      for (Integer shuffleId : shuffleIds) {
-        taskInfo.getCachedBlockIds().remove(shuffleId);
-        taskInfo.getCommitCounts().remove(shuffleId);
-        taskInfo.getCommitLocks().remove(shuffleId);
+      LOG.info("Start remove resource for appId[{}], shuffleIds[{}]", appId, shuffleIds);
+      final long start = System.currentTimeMillis();
+      final ShuffleTaskInfo taskInfo = shuffleTaskInfos.get(appId);
```

Comment:
> I think we need to limit the lock scope here.
> 
> `writeLock.lock()` should be started here rather than from the beginning of the method.

Cleaning up resources is not a frequent operation. It only happens when an `appId` expires and is removed, or when unregistering a `shuffleId`. Therefore, I don't think there's a need to narrow the scope of the lock, as it won't cause any performance bottlenecks. And the code will be more readable. 
Also, we have conducted performance stress tests on this PR and it does not impact performance.